File: 3138. Minimum Length of Anagram Concatenation.go
func minAnagramLength(s string) int {
	lenSqrt := int(math.Sqrt(float64(len(s))))
	answer := math.MaxInt32
	for i := 1; i <= lenSqrt; i++ {
		if len(s)%i != 0 {
			continue
		}
		if possible(s, i) {
			answer = min(answer, i)
		}
		if possible(s, len(s)/i) {
			answer = min(answer, len(s)/i)
		}
	}
	return answer
}

func possible(s string, length int) bool {
	freq := make([]int, 26)
	size := 0
	for i := 0; i < length; i++ {
		freq[int(s[i])-int('a')] += 1
		if freq[int(s[i])-int('a')] == 1 {
			size += 1
		}
	}
	for i := length; i < len(s); i += length {
		newFreq := make([]int, 26)
		matched := 0
		for j := i; j < i+length; j++ {
			newFreq[int(s[j])-int('a')] += 1
			if newFreq[int(s[j])-int('a')] == freq[int(s[j])-int('a')] {
				matched += 1
			}
		}
		if matched != size {
			return false
		}
	}
	return true
}

File: 3138. Minimum Length of Anagram Concatenation.java
class Solution {
    public int minAnagramLength(String s) {
        int answer = Integer.MAX_VALUE;
        for (int i = 1; i <= (int)Math.sqrt(s.length()); i++) {
            if (s.length() % i != 0) continue;
            if (possible(s, i)) answer = Math.min(answer, i);
            if (possible(s, s.length()/i)) answer = Math.min(answer, s.length()/i);
        }
        return answer;
    }
    private boolean possible(String s, int length) {
        int[] freq = new int[26];
        int size = 0;
        for (int i = 0; i < length; i++) {
            size = freq[s.charAt(i)-'a'] == 0 ? size + 1 : size;
            freq[s.charAt(i)-'a']++;
        }
        for (int i = length; i < s.length(); i += length) {
            int[] newFreq = new int[26];
            int matched = 0;
            for (int j = i; j < i+length; j++) {
                newFreq[s.charAt(j)-'a']++;
                if (newFreq[s.charAt(j)-'a'] == freq[s.charAt(j)-'a']) matched++;
            }
            if (matched != size) return false;
        }
        return true;
    }
}